template/template.go

```diff
@@ -74,30 +76,78 @@ func Named(s string) (*named, error) {
 	return nil, errors.New("no matching template found")
 }
 
+var DefaultTemplate, _ = Parse("{{ .Prompt }}")
+
 type Template struct {
 	*template.Template
 	raw string
 }
 
-func (t *Template) String() string {
-	return t.raw
+var response = parse.ActionNode{
+	NodeType: parse.NodeAction,
+	Pipe: &parse.PipeNode{
+		NodeType: parse.NodePipe,
+		Cmds: []*parse.CommandNode{
+			{
+				NodeType: parse.NodeCommand,
+				Args: []parse.Node{
+					&parse.FieldNode{
+						NodeType: parse.NodeField,
+						Ident:    []string{"Response"},
+					},
+				},
+			},
+		},
+	},
 }
 
-var DefaultTemplate, _ = Parse("{{ .Prompt }}")
-
 func Parse(s string) (*Template, error) {
-	t, err := template.New("").Option("missingkey=zero").Parse(s)
+	tmpl := template.New("").Option("missingkey=zero").Funcs(template.FuncMap{
+		"toJson": func(v any) string {
```

Comment:
nit: I've found that not using anonymous functions here makes these critical paths easier to test.

Ex:
```
func toJson(v any) string {
	b, err := json.Marshal(v)
	if err != nil {
		return ""
	}

	return string(b)
}

func isLastMessage(s []*api.Message, m *api.Message) bool {
	for i := len(s) - 1; i >= 0; i-- {
		if m.Role != s[i].Role {
			continue
		}

		return m == s[i]
	}

	return false
}

func Parse(s string) (*Template, error) {
	tmpl := template.New("").Option("missingkey=zero").Funcs(template.FuncMap{
		"toJson":        toJson,
		"isLastMessage": isLastMessage,
	})
```

Reply:
I pulled out the funcs so it can be unit tested but kept the functions themselves anonymous since they don't really have meaning outside of the template. I also replaced `isLastMessage` with `messages.Last(role, message)` which should be clearer